extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/AbstractOidcAuthenticationMechanism.java

```diff
@@ -4,18 +4,45 @@
 import io.quarkus.security.identity.IdentityProviderManager;
 import io.quarkus.security.identity.SecurityIdentity;
 import io.quarkus.security.identity.request.TokenAuthenticationRequest;
+import io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle;
 import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
 import io.quarkus.vertx.http.runtime.security.HttpSecurityUtils;
 import io.smallrye.mutiny.Uni;
+import io.vertx.core.Vertx;
 import io.vertx.ext.web.RoutingContext;
 
 abstract class AbstractOidcAuthenticationMechanism {
+    /**
+     * Propagate {@link TokenCredential} via Vert.X duplicated context if explicitly enabled and request context
+     * can not be activated.
+     */
+    private static final boolean PROPAGATE_TOKEN_CREDENTIAL_WITH_DUPLICATED_CTX = Boolean
+            .getBoolean("quarkus.oidc.propagate-token-credential-with-duplicated-context");
+    private static final String ERROR_MSG = "OIDC requires a safe (isolated) Vert.x sub-context for propagation of the '"
+            + TokenCredential.class.getName() + "', but the current context hasn't been flagged as such.";
     protected DefaultTenantConfigResolver resolver;
     private HttpAuthenticationMechanism parent;
 
     protected Uni<SecurityIdentity> authenticate(IdentityProviderManager identityProviderManager,
             RoutingContext context, TokenCredential token) {
         context.put(HttpAuthenticationMechanism.class.getName(), parent);
+
+        if (PROPAGATE_TOKEN_CREDENTIAL_WITH_DUPLICATED_CTX) {
```

Comment:
Can it be copied to quarkus-smallrye-jwt as well as it can work with the token propagation ?

Reply:
Added test as well.